src/lang/ArrayObject.php

```diff
@@ -164,12 +164,29 @@ public function slice(int $offset, ?int $length = null, bool $preserveKeys = fal
 	/**
 	 * Merges in other values
 	 *
-	 * @param mixed ... Variable list of arrays to merge.
+	 * @param mixed ...$toMerges Variable list of arrays to merge.
 	 *
 	 * @return ArrayObject $this
 	 */
-	public function merge(): self {
-		$this->array = array_merge($this->array, func_get_args());
+	public function merge(...$toMerges): self {
+		foreach ($toMerges as $toMerge) {
+			$this->array = array_merge($this->array, $toMerge);
```

Comment:
would `array_merge($this->array, ...$toMerges);` work here?

Reply:
Awesome catch!! I didn't understand that variadics could be used in such way. Thank you!

Fixed.